numpy/exceptions.py

```diff
@@ -0,0 +1,199 @@
+"""
+Exceptions and Warnings (:mod:`numpy.exceptions`)
+=================================================
+
+General exceptions used by NumPy.  Note that some exceptions may be module
+specific, such as linear algebra errors.
+
+.. versionadded:: NumPy 1.24
+
+    The exceptions module is new in NumPy 1.24.  Older exceptions remain
+    available through the main NumPy namespace for compatibility.
+
+.. currentmodule:: numpy.exceptions
+
+Warnings
+--------
+.. autosummary::
+   :toctree: generated/
+
+   ComplexWarning             Given when converting complex to real.
+   VisibleDeprecationWarning  Same as a DeprecationWarning, but more visible.
+
+Exceptions
+----------
+.. autosummary::
+   :toctree: generated/
+
+    AxisError       Given when an axis was invalid.
+    TooHardError    Error specific to `numpy.shares_memory`.
+
+"""
+
+
+from ._utils import set_module as _set_module
+
+__all__ = [
+    "ComplexWarning", "VisibleDeprecationWarning",
+    "TooHardError", "AxisError"]
+
+
+# Disallow reloading this module so as to preserve the identities of the
+# classes defined here.
+if '_is_loaded' in globals():
+    raise RuntimeError('Reloading numpy._globals is not allowed')
+_is_loaded = True
+
+
+# TODO: One day, we should remove the _set_module here before removing them
+#       fully.  Not doing it now, just to allow unpickling to work on older
+#       versions for a bit.  (Module exists since NumPy 1.24.)
+#       This then also means that the typing stubs should be moved!
+
+
+@_set_module('numpy')
+class ComplexWarning(RuntimeWarning):
+    """
+    The warning raised when casting a complex dtype to a real dtype.
+
+    As implemented, casting a complex number to a real discards its imaginary
+    part, but this behavior may not be what the user actually wants.
+
+    """
+    pass
+
+
+
+@_set_module("numpy")
+class ModuleDeprecationWarning(DeprecationWarning):
+    """Module deprecation warning.
+
+    .. warning::
+
+        This warning should not be used, since nose testing is not relvant
+        anymore.
```

Comment:
Added this warning, and skipped documenting it...  It seems like a remnant of a time past.